### PR TITLE
Switch to CMake build, upgrade to leveldb 1.22 (WIP)

### PR DESCRIPTION
**Pull request by chjj**
It seems that google has been switching to CMake for a few of their projects. They also added out of the box windows support to leveldb.

Along with upgrading leveldb, this PR switches to [cmake-node] for the build and finally rids us of the convoluted gyp build. leveldb and snappy's CMakeLists required only [minor changes] in order to get the build working.

The downside here is that users need to have CMake installed. This isn't _too_ different from requiring the presence of node-gyp: although node-gyp is bundled with npm and a shipped with a lot of node.js installs, it does require the user to have python installed on their system as well.

[cmake-node]: https://github.com/chjj/cmake-node
[minor changes]: https://github.com/bcoin-org/bdb/commit/d926b827d70293bd3fbf71707d6b4f6f4709b0cc

### CODE REVIEW

**Comment by tuxcanfly** [comment]
Looks good. Was able to build, but needed `cmake > 3.13` for `target_link_options`.

**Comment by tuxcanfly** [comment]
@chjj Would it be possible to cross compile for Windows (at least in theory)?

**Comment by chjj** [comment]
@tuxcanfly, wow, I had no idea `target_link_options` was so new. I changed cmake-node to fall back to `set_target_property(LINK_FLAGS)` if `target_link_options` isn't available.

> Would it be possible also use this for bcrypto and make it possible to cross compile bcoin for Windows (at least in theory)?

I guess it's possible CMake would have an easier time than gyp since you could use some kind of mingw32 toolchain file.

**Comment by tuxcanfly** [comment]
Thanks, tested with downgraded `cmake` `3.10.2` and works :white_check_mark:

**Comment by chjj** [comment]
After thinking about it, I feel like the question of using cmake is more of a meta-issue. For now I'm just going to merge this in with the new node-gyp binding. We can probably discuss switching all native addons to cmake later.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# CMakeLists.txt - cmake build for leveldown
+# Copyright (c) 2020, Christopher Jeffrey (MIT License).
+# https://github.com/bcoin-org/bdb
+
+cmake_minimum_required(VERSION 3.4)
+project(leveldown LANGUAGES CXX)
```
Comment (tuxcanfly) [inline CMakeLists.txt:6]: Might need `C CXX` here since we have a the `win_delay_load_hook.c` when building under Windows...

Comment (chjj) [reply CMakeLists.txt:6]: win_delay_load_hook.c is valid C++, so it seems to work without it. I think it's just building it as C++.

**Pull request merged**